drivers/entropy/entropy_ateccx08.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_ateccx08_entropy
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/entropy.h>
+#include <zephyr/drivers/eeprom/ateccx08.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(entropy_ateccx08, CONFIG_ENTROPY_LOG_LEVEL);
+
+struct entropy_ateccx08_config {
+	const struct device *parent;
+};
+
+struct entropy_ateccx08_data {
+	uint8_t random_buffer[32];
+	size_t buffer_pos;
+};
+
+static int entropy_ateccx08_get_entropy(const struct device *dev, uint8_t *buffer, uint16_t length)
+{
+	const struct entropy_ateccx08_config *config = dev->config;
+	struct entropy_ateccx08_data *dev_data = dev->data;
+
+	int ret = 0;
```

Comment:
This function is not re-entrant, you need to protect it otherwise you can have a concurrent call messing with it.